Rename pages as part of information architecture changes

niqjohnson opened the pull request:
Rename pages (either in the main navigation from the README, in their page titles, or both) to more closely match the content that's on them

## Changes

- Rename pages with titles that we identified as unclear
- Rename all the language standards pages to "{Language} Standards" so that they're consistent with the README navigation and with each other

## Testing

1. View this branch in GitHub
2. Make sure all the new page and navigation titles make sense

chosak commented on standards/css.md:
```diff
@@ -1,4 +1,4 @@
-# CSS & Less Guide
+# CSS & Less Standards
```
Minor thing, but in the README this page is listed as "CSS/Less" and here it's "CSS & Less". Personally I prefer the latter but it'd be good if they were consistent.

chosak commented:
@Scotchester one downside of doing that is anywhere that's linking to these files already, for example [here](https://cfpb.github.io/cfgov-refresh/testing-be/#linting) in the cfgov-refresh docs (although that particular link is broken because it refers to a branch that no longer exists). On the other hand, there might not be _that_ many of these links, and someone going to a broken link would (hopefully) easily be able to find their way back to the index.

Scotchester commented:
@chosak I considered that, but I think it's worth it. Those are some really bad filenames.

niqjohnson commented:
@Scotchester: Makes sense to me. I changed these file names:

- `guides/npm.md → guides/authoring-npm-modules.md`
- `guides/sauce-labs.md → guides/browser-testing-with-sauce-labs.md`
- `guides/javascript-modules-commonjs.md → guides/creating-commonjs-modules.md`
- `guides/testing.md → guides/front-end-testing.md`
- `guides/npm-usage.md → guides/using-npm-modules-in-projects.md`

I *think* I caught all the places these files were linked to in this repo and updated them to the new filenames, but let me know if you spot any I missed. We can update links in other repos once this is merged.
